framework/src/main/java/org/tron/core/config/args/Args.java

```diff
@@ -1521,22 +1521,12 @@ private static FilterQuery getEventFilter(final com.typesafe.config.Config confi
     return filter;
   }
 
-  private static void bindIp(final com.typesafe.config.Config config) {
-    if (!config.hasPath(Constant.NODE_DISCOVERY_BIND_IP)
-        || config.getString(Constant.NODE_DISCOVERY_BIND_IP)
-        .trim().isEmpty()) {
-      if (PARAMETER.nodeDiscoveryBindIp == null) {
-        logger.info("Bind address wasn't set, Punching to identify it...");
-        try (Socket s = new Socket("www.baidu.com", 80)) {
-          PARAMETER.nodeDiscoveryBindIp = s.getLocalAddress().getHostAddress();
-          logger.info("UDP local bound to: {}", PARAMETER.nodeDiscoveryBindIp);
-        } catch (IOException e) {
-          logger.warn("Can't get bind IP. Fall back to 127.0.0.1: " + e);
-          PARAMETER.nodeDiscoveryBindIp = "127.0.0.1";
-        }
-      }
-    } else {
-      PARAMETER.nodeDiscoveryBindIp = config.getString(Constant.NODE_DISCOVERY_BIND_IP).trim();
+  private static void bindIp() {
+    try (Socket s = new Socket("www.baidu.com", 80)) {
+      PARAMETER.nodeDiscoveryBindIp = s.getLocalAddress().getHostAddress();
+    } catch (IOException e) {
+      logger.warn("Can't get bind IP. Fall back to 127.0.0.1: " + e);
+      PARAMETER.nodeDiscoveryBindIp = "127.0.0.1";
```

Comment:
Why should this logic be retained?

Reply:
if the external ip/W is not exist, use LAN IP.

Reply:
Why hardcoded into `www.baidu.com`?

Reply:
@halibobo1205 It has been moved to libp2p module. This url is not necessary and we can use alternative way.